File: pinyougou_parent/pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;
import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
/**
 * 业务逻辑接口
 * @author Steven
 *
 */
public interface BrandService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbBrand> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(TbBrand brand);
	
	
	/**
	 * 修改
	 */
	public void update(TbBrand brand);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbBrand findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TbBrand brand, int pageNum, int pageSize);

	/**
	 * 商家品牌查询：查询+分页（区别在于商家查询只返回自己所申请过的品牌）
	 * @param loginName 当前商家名
	 * @return
	 */
	public List<TbBrand> sellerFindAll(String loginName);

	/**
	 * 商家品牌查询：查询+分页（区别在于商家查询只返回自己所申请过的品牌）
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult sellerFindPage(TbBrand brand, int pageNum, int pageSize,String sellerId);



	/**
	 * 审核状态
	 * @param ids
	 * @param status
	 */
    void updateStatus(Long[] ids, Integer status);

	/**通过name找品牌的id
	 * @param name
	 * @return
	 */
    Long findBrandIdByName(String name);
}

File: pinyougou_parent/pinyougou_shop_web/src/main/webapp/js/service/brandService.js
//服务层
app.service('brandService',function($http){
    // //给模板模块查询所有的品牌列表调用
    // this.getAll=function(){
    //     return $http.get('../brand/findAll.do');
    // }
	//读取列表数据绑定到表单中
	this.findAll=function(loginName){
		return $http.get('../brand/findAll.do?loginName='+loginName);
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../brand/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../brand/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity,loginName){
		return  $http.post('../brand/sellerAdd.do?loginName='+loginName,entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../brand/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../brand/delete.do?ids='+ids);
	}

    //商家自家的品牌搜索
	this.search=function(sellerId,page,rows,searchEntity){
        return $http.post('../brand/sellerSearch.do?sellerId='+sellerId+'&page='+page+"&rows="+rows, searchEntity);
    }
});
